src/app/pages/office/office.analytics.effects.ts

```diff
@@ -0,0 +1,326 @@
+        action.errorMessage);
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  completeTest$ = createEffect(() => this.actions$.pipe(
+    ofType(CompleteTest),
+    concatMap((action) => of(action).pipe(
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(getRekeyIndicator),
+          select(isRekey),
+        ),
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(getJournalData),
+          select(getCandidate),
+          select(getCandidateId),
+        ),
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(getJournalData),
+          select(getApplicationReference),
+          select(getApplicationNumber),
+        ),
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(isPassed),
+        ),
+      ),
+    )),
+    switchMap((
+      [, isTestRekey, candidateId, applicationReference, isTestPassed]:
+      [ReturnType<typeof CompleteTest>, boolean, number, string, boolean],
+    ) => {
+
+      this.analytics.addCustomDimension(AnalyticsDimensionIndices.CANDIDATE_ID, `${candidateId}`);
+      this.analytics.addCustomDimension(AnalyticsDimensionIndices.APPLICATION_REFERENCE, applicationReference);
+
+      this.analytics.logEvent(
+        AnalyticsEventCategories.POST_TEST,
+        isTestRekey ? AnalyticsEvents.COMPLETE_REKEY_TEST : AnalyticsEvents.COMPLETE_TEST,
+        isTestPassed ? 'pass' : 'fail',
+      );
+
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  setCircuit$ = createEffect(() => this.actions$.pipe(
+    ofType(CircuitTypeChanged),
+    concatMap((action) => of(action).pipe(
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+        ),
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(getTestCategory),
+        ),
+      ),
+    )),
+    concatMap(([action, tests, category]: [ReturnType<typeof CircuitTypeChanged>, TestsModel, CategoryCode]) => {
+      this.analytics.addCustomDimension(AnalyticsDimensionIndices.TEST_CATEGORY, category);
+      this.analytics.logEvent(
+        formatAnalyticsText(AnalyticsEventCategories.OFFICE, tests),
+        formatAnalyticsText(AnalyticsEvents.CIRCUIT_CHANGED, tests),
+        `Circuit type ${action.circuitType} selected`,
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  setIndependentDrivingType$ = createEffect(() => this.actions$.pipe(
+    ofType(IndependentDrivingTypeChanged),
+    concatMap((action) => of(action).pipe(
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+        ),
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(getTestCategory),
+        ),
+      ),
+    )),
+    // eslint-disable-next-line max-len
```

Comment:
can we remove this and just press return after the ```:``` on the next line?

Reply:
agree. far to many of these already.